base_rest_demo/tests/test_new_api.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2021 Wakari SRL
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import odoo.tests.common
+from odoo.tests import new_test_user
+from odoo.tests.common import HttpCase
+
+HOST = "127.0.0.1"
+PORT = odoo.tools.config["http_port"]
+
+
+@odoo.tests.common.at_install(False)
+@odoo.tests.common.post_install(True)
+class TestPartnerNewAPI(HttpCase):
+    def setUp(self):
+        super().setUp()
+        self.partner_data = {
+            "name": "Test Partner",
+            "country_id": self.env["res.country"].search([], limit=1).id,
+            "state_id": self.env["res.country.state"].search([], limit=1).id,
+        }
+        self.test_partner = self.env["res.partner"].create(self.partner_data)
+        self.password = "test_password"
+        self.user = new_test_user(
+            self.env,
+            login="test_user",
+            password=self.password,
+            groups="base.group_user",
+        )
+
+    @odoo.tools.mute_logger("odoo.addons.base_rest.http")
+    def test_01(self):
```

Comment:
Your test should not be an HttpCase test. Its possible to test the `restapi.Datamodel`class alone...
For ex:
```python
import mock
from odoo.addons.datamode import fields
from odoo.addons.datamode.core import Datamodel
from odoo.addons.datamodel.tests import common
from odoo.addons.base_rest_data_model import restapi



class TestDataModel(common.DatamodelRegistryCase):

    def test_to_response_exclude(self):
        class MyModel(DataModel):
            _name = "my.model"
            ....

        MyModel._build_datamodel(self.datamodel_registry)

        restapi_datamodel = restapi.Datamodel("my.model")
        mock_service = mock.Mock()
        mock.env.datamodels = self.datamodel_registry

        my_model = MyModel()
        res = restapi_datamodel.to_response(mock_service, my_model)
        ...
```
This test can be therefore put into the base_rest_datamodel addon.